bytecomp/translmod.ml

```diff
@@ -839,21 +852,22 @@ let field_of_str loc str =
 
 
 let transl_store_structure glob map prims str =
+  let no_env_update _ _ env = env in
```

Comment:
If the implementation of substitution on environments is adapted to remove bindings before `update_env` is called, then this function could/should renamed into `remove_from_env`, and I believe this would be the correct behavior for the use-sites below.